### ISSUE TITLE
File download button needed in submitted file modal dialogs

### ISSUE THREAD

**Issue by Mankro** [apluslms/a-plus#894]
(By downloading, I mean that the browser downloads the file to the user's computer. Firefox usually shows a dialog while Chrome usually downloads it immediately. The A+ submitted file URL already supports the parameter `?download=yes`.) 

The student can see file contents in some dialogs, but there is no button for downloading the file to the user's computer. A download button would be very good to have in some places: 

* model solution modal dialog in chapter exercises,
* model solution page in the full-view exercise,
* submission template/skeleton code modal dialog in chapter exercises, 
* submission template file in the full-view exercise page,
* submitted file modal dialog in the submission modal in chapter exercises and
* the submitted file modal dialog in full-view submission pages outside chapters

_Originally posted by @Mankro in https://github.com/apluslms/a-plus/pull/893#discussion_r738336787_

PR #893 already has a quick implementation for the download button in the inspect submission page. Other places need the download button too.

**Comment by Mankro** [comment]
From #485:

> > **Migrated from Trello**
> 
> In A+, the exercise template files (e.g., skeleton code for the student in a programming exercise) functionality currently only shows the content of the files in a modal dialog. The user may not directly download the file. A+ should enable users to download the files too instead of just copy-pasting from the modal dialog.
> 
> The modal could, for example, have a download button that downloads the actual file to the student's computer. This download URL could use some GET query parameter (?download=yes) so that the URL is almost the same as what is used for the current modal.
> 
> **Comments**
> 
> > **Jaakko Kantojärvi:**
> > As extra, add a feature to download all template files as a zip (tar.gz can be created in memory, maybe zips can be created too)
> 
> > **Jaakko Kantojärvi:**
> > The modal could, for example, have a download button that downloads the actual file to the student's computer. This download URL could use some GET query parameter `(?download=yes)` so that the URL is almost the same as what is used for the current modal.

**Comment by Mankro** [comment]
> submitted file modal dialog in the submission modal in chapter exercises and
the submitted file modal dialog in full-view submission pages outside chapters

These weren't considered in the pull request #1019. The JS code (`highlightCode()`) adds the new download buttons to those as well, but the button is broken because the `data-url` attribute is missing from the html.

**Comment by Mankro** [comment]
> > submitted file modal dialog in the submission modal in chapter exercises and
> > the submitted file modal dialog in full-view submission pages outside chapters
> 
> These weren't considered in the pull request #1019. The JS code (`highlightCode()`) adds the new download buttons to those as well, but the button is broken because the `data-url` attribute is missing from the html.

Fixed in https://github.com/apluslms/a-plus/commit/9124fc7938a8cfc5d5f2a71f3f925c1cf5457237
